packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts

```diff
@@ -33,10 +33,449 @@ export interface IVirtualNode extends cdk.IResource {
+    return new GrpcNodeListener(props);
+  }
+
+  /**
+   * Returns an TCP Listener for a VirtualNode
+   */
+  public static tcpNodeListener(props?: TcpNodeListenerProps): VirtualNodeListener {
+    return new TcpNodeListener(props);
+  }
+
+  /**
+  * Binds the current object when adding Listener to a VirtualNode
+  */
+  public abstract bind(scope: cdk.Construct): CfnVirtualNode.ListenerProperty;
+}
+
+/**
+ * Represents the properties required to define a HTTP Listener for a VirtualNode.
+ */
+export class HttpNodeListener extends VirtualNodeListener {
+/**
+ * Returns the ListenerTimeoutProperty for HTTP protocol
+ */
+  public static renderTimeout(tm: HttpTimeout): CfnVirtualNode.ListenerTimeoutProperty {
+    return ({
+      http: {
+        idle: tm?.idle !== undefined ? {
+          unit: 'ms',
+          value: tm?.idle.toMilliseconds(),
+        } : undefined,
+        perRequest: tm?.perRequest !== undefined ? {
+          unit: 'ms',
+          value: tm?.perRequest.toMilliseconds(),
+        } : undefined,
+      },
+    });
+  }
+
+  /**
+   * Port to listen for connections on
+   *
+   * @default - 8080
+   */
+  readonly port: number;
+
+  /**
+   * Health checking strategy upstream nodes should use when communicating with the listener
+   *
+   * @default - no healthcheck
+   */
+  readonly healthCheck?: HealthCheck;
+
+  /**
+   * Listener timeout property for HTTP protocol
+   *
+   * @default - none
+   */
+  readonly timeout?: HttpTimeout;
+
+  constructor(props?: HttpNodeListenerProps) {
+    super();
+    const checkedProps = props ?? {};
+    this.port = checkedProps.port ? checkedProps.port : 8080;
+    this.healthCheck = checkedProps.healthCheck;
+    this.timeout = checkedProps.timeout;
+  }
+
+  /**
+   * Return Listener for HTTP protocol when Listener is added to Virtual Node.
+   */
+  public bind(_scope: cdk.Construct): CfnVirtualNode.ListenerProperty {
+    return {
+      portMapping: {
+        port: this.port,
+        protocol: Protocol.HTTP,
+      },
+      healthCheck: HttpNodeListener.renderHealthCheck({
+        port: this.port,
+        protocol: Protocol.HTTP,
+      }, this.healthCheck),
+      timeout: this.timeout ? HttpNodeListener.renderTimeout(this.timeout) : undefined,
+    };
+  }
+}
+
+/**
+ * Represents the properties required to define a HTTP2 Listener for a VirtualNode.
+ */
+export class Http2NodeListener extends VirtualNodeListener {
+/**
+ * Returns the ListenerTimeoutProperty for HTTP2 protocol
+ */
+  public static renderTimeout(tm: Http2Timeout): CfnVirtualNode.ListenerTimeoutProperty {
+    return ({
+      http2: {
+        idle: tm?.idle !== undefined ? {
+          unit: 'ms',
+          value: tm?.idle.toMilliseconds(),
+        } : undefined,
+        perRequest: tm?.perRequest !== undefined ? {
+          unit: 'ms',
+          value: tm?.perRequest.toMilliseconds(),
+        } : undefined,
+      },
+    });
+  }
```

Comment:
All of this is copy-pasted from `HttpNodeListener`.

Perhaps `HttpNodeListener` and `Http2NodeListener` should share the same `timeout` type?